src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -142,12 +142,18 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
           }
           break;
         case FORMAT_NAME_NODES :
-          if (!liveState.isNameNode1Initialized()) {
+          if (!liveState.isNameNode1Initialized() && !liveState.isNameNode2Initialized()) {
             sendMessageTo(
                 driver,
                 liveState.getFirstNameNodeTaskId(),
                 liveState.getFirstNameNodeSlaveId(),
                 HDFSConstants.NAME_NODE_INIT_MESSAGE);
+          } else if (!liveState.isNameNode1Initialized()) {
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(),
+                liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
```

Comment:
Yes, bootstrap works properly. The reason is that after one NN is killed, the other one becomes active. Then you bootstrap the second NN off the active NN to get the latest information. Bootstrap will only be called if there is already one active NN running.